src/wp-includes/media.php

```diff
@@ -1980,6 +2023,30 @@ function wp_image_use_alternate_mime_types( $image, $context, $attachment_id ) {
 	return $image;
 }
 
+/**
+ * Helper function find the smallest image from a given sizes sources array.
+ *
+ * @param array $sources The media meta sizes->sources array.
+ * @param string $target_file The initial target file to find the smallest file for.
+ * @return array The smallest image size in the sources array.
+ */
+function _wp_get_smallest_file_from_sources( $sources, $target_file ) {
+	// Choose the smallest image size available.
+	if ( empty ( target_file['filesize'] ) ) {
+		return $target_file;
+	}
+
+	foreach( $metadata['sources'] as $source ) {
```

Comment:
possibly add a check for a non empty array before the forloop.

Also, the variable should possibly be `$sources` instead of the `$metadata`